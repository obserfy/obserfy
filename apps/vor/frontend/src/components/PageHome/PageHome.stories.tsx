import { FC } from "react"
import PageHome from "./PageHome"

export default {
  title: "Page/PageHome",
  component: PageHome,
  parameters: {
    componentSubtitle: "Just a simple PageHome",
  },
}

export const Basic: FC = () => <PageHome />
