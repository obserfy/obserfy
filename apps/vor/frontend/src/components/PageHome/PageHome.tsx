import { FC } from "react"
import { Box } from "theme-ui"
import { useBreakpointIndex } from "@theme-ui/match-media"
import StudentsList from "../StudentsList/StudentsList"

export const PageHome: FC = () => {
  const breakpoint = useBreakpointIndex({ defaultIndex: 0 })

  if (breakpoint > 1) {
    return <></>
  }

  return (
    <Box sx={{ display: ["block", "block", "none"] }} data-cy="mobile-students">
      <StudentsList />
    </Box>
  )
}

export default PageHome
