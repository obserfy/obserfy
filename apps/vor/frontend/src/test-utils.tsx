/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
