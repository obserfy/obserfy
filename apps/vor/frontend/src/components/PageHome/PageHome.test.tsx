import { render } from "../../test-utils"
import { Basic } from "./PageHome.stories"

describe("PageHome", () => {
  it("should render correctly", () => {
    // TODO: This is a temporary fix, make sure to test mock response correctly.
    const response: string[] = []
    fetchMock.mockResponse(JSON.stringify(response))

    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
