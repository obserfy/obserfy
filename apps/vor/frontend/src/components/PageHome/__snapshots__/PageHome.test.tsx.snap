// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHome should render correctly 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 0.05;
  }

  50% {
    opacity: 0.2;
  }

  100% {
    opacity: 0.05;
  }
}

@keyframes animation-0 {
  0% {
    opacity: 0.05;
  }

  50% {
    opacity: 0.2;
  }

  100% {
    opacity: 0.05;
  }
}

@keyframes animation-0 {
  0% {
    opacity: 0.05;
  }

  50% {
    opacity: 0.2;
  }

  100% {
    opacity: 0.05;
  }
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
}

@media screen and (min-width: 40em) {
  .emotion-1 {
    display: block;
  }
}

@media screen and (min-width: 52em) {
  .emotion-1 {
    display: none;
  }
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-bottom: 32px;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 16px;
  padding-top: 16px;
  padding-bottom: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: var(--theme-ui-colors-textMediumEmphasis);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 13px;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  margin-right: -33px;
  margin-left: 8px;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
  width: 24px;
  height: 24px;
  z-index: 2;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  width: auto;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 12px;
  color: var(--theme-ui-colors-text);
  padding-top: 8px;
  padding-bottom: 8px;
  margin-right: 8px;
  background-color: var(--theme-ui-colors-surface);
  --theme-ui-input-autofill-bg: var(--theme-ui-colors-background);
  padding-left: 40px;
  font-size: 14px;
  width: 100%;
  width: 100%;
  width: 100%;
}

.emotion-7:autofill,
.emotion-7:autofill:hover,
.emotion-7:autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-7:autofill:first-line,
.emotion-7:autofill:hover:first-line,
.emotion-7:autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-7:-webkit-autofill,
.emotion-7:-webkit-autofill:hover,
.emotion-7:-webkit-autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-7:-webkit-autofill:first-line,
.emotion-7:-webkit-autofill:hover:first-line,
.emotion-7:-webkit-autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-7:hover,
.emotion-7:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-7::-webkit-input-placeholder {
  font-style: italic;
}

.emotion-7::-moz-placeholder {
  font-style: italic;
}

.emotion-7:-ms-input-placeholder {
  font-style: italic;
}

.emotion-7::placeholder {
  font-style: italic;
}

.emotion-7:disabled {
  opacity: 0.68;
  border-color: var(--theme-ui-colors-border);
  color: var(--theme-ui-colors-textDisabled);
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: var(--theme-ui-colors-surface);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  padding-left: 8px;
  padding-right: 8px;
  height: 100%;
}

.emotion-8:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-8:hover,
.emotion-8:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-8:hover:disabled,
.emotion-8:focus:disabled {
  background-color: transparent;
  border-color: var(--theme-ui-colors-border);
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
}

.emotion-10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 16px;
  padding-right: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-11 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-right: 8px;
  margin-bottom: 8px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: var(--theme-ui-colors-primary);
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  border-radius: 6px;
  border-color: var(--theme-ui-colors-border);
  border-width: 1px;
  border-style: solid;
  white-space: nowrap;
}

.emotion-11:hover {
  background-color: var(--theme-ui-colors-primary);
}

.emotion-12 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  color: var(--theme-ui-colors-onPrimary);
  font-size: 14px;
}

.emotion-13 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-14 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-top: 8px;
  padding-bottom: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-15 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-radius: 6px;
  -webkit-animation: 4s infinite animation-0;
  animation: 4s infinite animation-0;
  background-color: #a5a5a5;
  width: 32px;
  height: 32px;
  border-radius: 99999px;
}

.emotion-16 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: 16px;
  margin-right: auto;
  border-radius: 6px;
  -webkit-animation: 4s infinite animation-0;
  animation: 4s infinite animation-0;
  background-color: #a5a5a5;
  width: 10rem;
  height: 22px;
}

.emotion-17 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-radius: 6px;
  -webkit-animation: 4s infinite animation-0;
  animation: 4s infinite animation-0;
  background-color: #a5a5a5;
  width: 4rem;
  height: 22px;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
      data-cy="mobile-students"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          <label
            class="emotion-4"
          >
            <div
              class="emotion-5"
            >
              <div
                class="emotion-6"
              />
              <input
                class="emotion-7"
                name="name"
                placeholder="Search students"
                value=""
              />
            </div>
          </label>
          <a
            href="/dashboard/students/new"
            style="flex-shrink: 0;"
          >
            <button
              class="emotion-8"
              data-cy="addStudent"
            >
              <div
                class="emotion-9"
              />
            </button>
          </a>
        </div>
        <div
          class="emotion-10"
        >
          <div
            class="emotion-11"
          >
            <p
              class="emotion-12"
            >
              All
            </p>
          </div>
        </div>
        <div
          class="emotion-13"
        >
          <div
            class="emotion-14"
          >
            <div
              class="emotion-15"
            />
            <div
              class="emotion-16"
            />
            <div
              class="emotion-17"
            />
          </div>
          <div
            class="emotion-14"
          >
            <div
              class="emotion-15"
            />
            <div
              class="emotion-16"
            />
            <div
              class="emotion-17"
            />
          </div>
          <div
            class="emotion-14"
          >
            <div
              class="emotion-15"
            />
            <div
              class="emotion-16"
            />
            <div
              class="emotion-17"
            />
          </div>
          <div
            class="emotion-14"
          >
            <div
              class="emotion-15"
            />
            <div
              class="emotion-16"
            />
            <div
              class="emotion-17"
            />
          </div>
          <div
            class="emotion-14"
          >
            <div
              class="emotion-15"
            />
            <div
              class="emotion-16"
            />
            <div
              class="emotion-17"
            />
          </div>
          <div
            class="emotion-14"
          >
            <div
              class="emotion-15"
            />
            <div
              class="emotion-16"
            />
            <div
              class="emotion-17"
            />
          </div>
          <div
            class="emotion-14"
          >
            <div
              class="emotion-15"
            />
            <div
              class="emotion-16"
            />
            <div
              class="emotion-17"
            />
          </div>
          <div
            class="emotion-14"
          >
            <div
              class="emotion-15"
            />
            <div
              class="emotion-16"
            />
            <div
              class="emotion-17"
            />
          </div>
          <div
            class="emotion-14"
          >
            <div
              class="emotion-15"
            />
            <div
              class="emotion-16"
            />
            <div
              class="emotion-17"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
